src/components/assignments/AssignmentForm.jsx

```diff
@@ -0,0 +1,107 @@
+import NumberInput from 'grommet/components/NumberInput';
+import CheckBox from 'grommet/components/CheckBox';
+import Button from 'grommet/components/Button';
+import Heading from 'grommet/components/Heading';
+import Footer from 'grommet/components/Footer';
+import Paragraph from 'grommet/components/Paragraph';
+
+const AssignmentForm = (props) => {
+  return (
+    <Form onSubmit={props.onSubmit} pad="medium">
+      <Heading tag="h2">{props.heading}</Heading>
+      <fieldset>
+        <legend><Paragraph size="small">Input the assignment details</Paragraph></legend>
+        <FormField htmlFor="name" label="Name" help="required">
+          <TextInput
+            id="name"
+            placeHolder="Assignment name"
+            onDOMChange={props.onChange}
+            required={true}
+            value={props.fields.name}
+          />
+        </FormField>
+        <FormField htmlFor="description" label="Description" help="required">
+          <textarea
+            id="description"
+            placeholder="Assignment description"
+            onChange={props.onChange}
+            required={true}
+            value={props.fields.description}
+          />
+        </FormField>
+        <FormField htmlFor="duedate" label="Due date" help="required">
+          <DateTime
+            id="duedate"
+            onChange={props.onChangeDate}
+            required={true}
+            value={props.fields.duedate}
+          />
+        </FormField>
+        <FormField htmlFor="classifications_target" label="Number of images per student" help="required">
+          <NumberInput
+            id="classifications_target"
+            min={1}
+            onChange={props.onChange}
+            required={true}
+            value={props.fields.classifications_target}
+          />
+        </FormField>
+      </fieldset>
+
+      <fieldset>
+        <legend><Paragraph size="small">Select the students to assign</Paragraph></legend>
+        {props.students.length === 0 &&
+          <Paragraph>There are no students available to assign.</Paragraph>}
+        {props.students.length > 0 &&
+          props.students.map((student) => {
+            <CheckBox key={student.id} label="A student user" onChange={() => {}} />
+          })}
+        <Button onClick={props.students.length > 0 ? () => {} : null} label="Select all students" />
+      </fieldset>
+
+      <fieldset>
+        <legend><Paragraph size="small">You need to select images for this assignment. Click "Select images" below, then use the map to choose a set of images for your students to identify and click "Select for assignment".</Paragraph></legend>
+        <Button onClick={() => { alert('Subject selection modal!')}} label="Select images" />
+      </fieldset>
+      <Footer>
+        <Button type="submit" label={props.submitLabel} primary={true} />
+      </Footer>
+    </Form>
+  );
+};
+
+AssignmentForm.defaultProps = {
+  onChange: () => {},
+  fields: {
+    name: '',
+    subject: '',
```

Comment:
I'm not sure if this is supposed to be `fields.subject` and `fields.school`; I think we need default props for `fields.name|description|duedate|classifications_target`

Reply:
Yep, probably cruft from when I copied and pasted from the classroom's form as a place to start.